### ISSUE TITLE
Better documentation

### ISSUE THREAD

**Issue by seiyria** [mankindsoftware/angular-isotope#18]
Hi,

Could you please provide better documentation on the directive options as well as having the code used to create examples directly in the examples? It saves me having to go digging through the code to find what I can use. Thanks!

**Comment by anderskristo** [comment]
Agreed, would be nice with some deeper documentation. Such as imagesLoaded and Masonry layout examples.

**Comment by mankindsoftware** [comment]
I just updated the demo with documentation at http://mankindsoftware.github.io/angular-isotope/.

I've worked under the assumption that David DeSandro's Isotope, with documention at http://isotope.metafizzy.co/, is the real hero and angular-isotope, being pretty much just a wrapper, is a side-kick wanna-be. ImagesLoaded and Masonry layout examples are given at the Isotope site.

I hope that event-based access through Angular-Isotope of the Isotope options and methods will cover a lot of bases.

Thanks for the feedback!
Mark

**Comment by anderskristo** [comment]
@mankindsoftware i've added imageloaded to my app and it seems to work, but it seems that isotope is running before my code in my controller.

This is my imagesloaded code from my controller:

```
     angular.element('#isotopeContainer').imagesLoaded().always((instance) ->
        console.log "all images loaded"
        return
      ).done((instance) ->
        console.log "all images successfully loaded"            
        return
      ).fail(->
        console.log "all images loaded, at least one is broken"
        return
      ).progress (instance, image) ->
        result = (if image.isLoaded then "loaded" else "broken")
        console.log "image is " + result + " for " + image.img.src
        return
```

What i want is when the .done is successful i want to run isotope, how?

**Comment by gablabelle** [comment]
@mankindsoftware I have to agree with them. Don't get me wrong, I really appreciate your work but for the newbie coder that I am, seeing a working example of how "normal" isotope options translate to the iso-options-subscribe="my-iso-opts" and/or iso-method-subscribe="my-iso-method" that your directives use would greatly help.

You say that ImagesLoaded and Masonry layout examples are given at the Isotope site. True, but how do I use it with angular-isotope? I'm stuck. Example would greatly appreciated. Many thanks.

**Comment by designbyadrian** [comment]
There are many bits and pieces missing. Some people seem to forget to add the directive to the app properly, for instance.

Also, there is no doc on how to use Metafizzy's extra parameters with your directive.

And what is this "extra" element that comes up in your examples?

**Comment by Emerson** [comment]
Would love to see examples of how to pass options (not using $emit)
